File: app/src/main/java/com/nanchen/aiyagirl/module/web/WebPresenter.java
package com.nanchen.aiyagirl.module.web;

import android.app.Activity;
import android.content.Intent;

import com.nanchen.aiyagirl.module.web.WebContract.IWebPresenter;
import com.nanchen.aiyagirl.module.web.WebContract.IWebView;

/**
 * Author: nanchen
 * Email: liushilin520@foxmail.com
 * Date: 2017-04-14  14:41
 */

public class WebPresenter implements IWebPresenter {
    private IWebView mWebView;
    private String mGankUrl;
    private Activity mActivity;

    public WebPresenter(IWebView webView){
        this.mWebView = webView;
    }


    @Override
    public void subscribe() {
        mActivity = mWebView.getWebViewContext();
        Intent intent = mActivity.getIntent();
        mWebView.setGankTitle(intent.getStringExtra(WebViewActivity.GANK_TITLE));
        mWebView.initWebView();
        mGankUrl = intent.getStringExtra(WebViewActivity.GANK_URL);
        mWebView.loadGankUrl(mGankUrl);
    }

    @Override
    public void unSubscribe() {

    }

    @Override
    public String getGankUrl() {
        return this.mGankUrl;
    }
}

File: app/src/main/java/com/nanchen/aiyagirl/module/web/WebPresenter.kt
package com.nanchen.aiyagirl.module.web

import android.app.Activity
import com.nanchen.aiyagirl.module.web.WebContract.IWebPresenter
import com.nanchen.aiyagirl.module.web.WebContract.IWebView

/**
 * Author: nanchen
 * Email: liushilin520@foxmail.com
 * Date: 2017-04-14  14:41
 */

class WebPresenter(private val mWebView: IWebView) : IWebPresenter {
    override lateinit var gankUrl: String
    private var mActivity: Activity? = null


    override fun subscribe() {
        mActivity = mWebView.webViewContext
        val intent = mActivity!!.intent
        mWebView.setGankTitle(intent.getStringExtra(WebViewActivity.GANK_TITLE))
        mWebView.initWebView()
        gankUrl = intent.getStringExtra(WebViewActivity.GANK_URL)
        mWebView.loadGankUrl(gankUrl)
    }


    override fun unSubscribe() {

    }
}
